echopype/clean/utils.py

```diff
@@ -83,6 +84,95 @@ def calc_transient_noise_pooled_Sv(
     return pooled_Sv
 
 
+def index_binning_pool_Sv(
+    ds_Sv: xr.Dataset,
+    func: str,
+    depth_bin: int,
+    num_side_pings: int,
+    exclude_above: float,
+    chunk_dict: dict,
+) -> xr.DataArray:
+    """
+    Compute pooled Sv array for transient noise masking using index binning.
+
+    This function makes the assumption that within each channel, the difference
+    between depth values is uniform across all pings. Thus, computing the number of
+    range sample indices needed to cover the depth bin is a channel-specific task.
+    """
+    # Create `ds_Sv` copy
+    ds_Sv_copy = (
+        ds_Sv.copy()
+        # Drop `filenames` dimension if exists and transpose Dataset
+        .drop_dims("filenames", errors="ignore").transpose("channel", "ping_time", "range_sample")
+    )
+
+    # Compute number of range sample indices that are needed to encapsulate the `depth_bin`
+    # value per channel.
+    all_chan_num_range_sample_indices = np.ceil(
+        depth_bin / np.nanmean(np.diff(ds_Sv_copy["depth"], axis=2), axis=(1, 2))
+    ).astype(int)
+
+    # Create list for pooled Sv DataArrays
+    pooled_Sv_list = []
+
+    # Iterate through channels
+    for channel_index in range(len(ds_Sv_copy["channel"])):
+        # Create calibrated Sv DataArray copies and remove values too close to the surface
+        min_range_sample = (ds_Sv["depth"] <= exclude_above).argmin().values
+        chan_Sv = (
+            ds_Sv_copy["Sv"]
+            .copy()
+            .isel(
+                channel=channel_index,
+                range_sample=slice(min_range_sample, len(ds_Sv["range_sample"])),
+            )
+        )
+        chan_pooled_Sv = (
+            ds_Sv_copy["Sv"]
+            .copy()
+            .isel(
+                channel=channel_index,
+                range_sample=slice(min_range_sample, len(ds_Sv["range_sample"])),
+            )
+        )
+
+        # Grab channel-specific number of range sample indices for vertical binning
+        chan_num_range_sample_indices = all_chan_num_range_sample_indices[channel_index]
+
+        # Create pooling size list
+        pooling_size = [(2 * num_side_pings) + 1, (2 * chan_num_range_sample_indices) + 1]
```

Comment:
Here you have a discrepancy of 1* vs 2* of the value in `chan_num_range_sample_indices`. Can you make it consistent such that the number of pixels used is actually the closest integer to what the user wants from the calling function?

Reply:
Sorry, I'm not quite understanding this.

So here's an example of what I intended the code to do:
If there is a 0.2m separation between each range sample index and the user wants a depth bin of 1m, then I calculate this `chan_num_range_sample_indices` value to be 5, since it takes 5 values to capture it. I pick a single point, and I go up 5 indices and down 5 indices of it. That single point is this `+1` term and the up and down is this `2 * chan_num_range_sample_indices` term.